### PR TITLE
feat: second assignment

### PR DESCRIPTION
**Pull request by JulianaCotabarren**
The second assignment was achieved as a new feature of the first one.
You will find the "OTHER HEROES" link on the navbar wich redirects to a new section where you could visualize all the Marvel Heroes from the json file. 
Also, there is a button (Load Sorted Heroes) that allows the replacement of the unordered heroes with the ordered array.

### CODE REVIEW

File: Juliana/second-assignment/script.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+     
+    heroesList.appendChild(heroeName);
+    heroesList.appendChild(heroeRelease);
+    heroesList.appendChild(heroeLink);
+
+    heroesContainer.appendChild(heroesList);        
+})
+}
+
+function showSortedHeroes(){
```
Comment (mhechavarria) [inline Juliana/second-assignment/script.js:50]: Every time you try to show sorted heroes it loads everything again and add it to the existing data, it should work only with the data that loads the first time.

File: Juliana/second-assignment/script.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    arrHeroes.forEach((item)=>{
+    let heroesList = document.createElement('div');
+    let heroeName = document.createElement('h2');
+    let heroeRelease = document.createElement('p');
+    let heroeLink = document.createElement('a');
+
+    const dateStr = item.modified,        
+    [yyyy,mm,dd] = dateStr.split(/[/:\-T]/)
```
Comment (mhechavarria) [inline Juliana/second-assignment/script.js:58]: Same here, try to use the Date object to format your release date as you want.

**Review by mhechavarria — CHANGES_REQUESTED** [review]
Overall looks really good, just left a couple of comments on things you can improve. One more thing if you are up to it, you can try to refactor the code so all the functions don't have side effects. Try to pass the array as a parameter to the functions and work with a copy of it instead of modifying it directly inside of every function.

File: Juliana/second-assignment/script.js (hunk 1 above)
Comment (JulianaCotabarren) [reply Juliana/second-assignment/script.js]: I fixed this issue in commit [c657b43]

File: Juliana/second-assignment/script.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    showSortedData();
+}
+
+function undoReplaceHeroes(){
+    sortedHeroesContainer.replaceWith(heroesContainer);
+    showData();
+}
+
+function sortHeroes(){
+    const sortHeroes = (x,y) => x.modified.localeCompare(y.modified);
```
[the remark this answers is not in the document]
Comment (JulianaCotabarren) [reply Juliana/second-assignment/script.js:55]: I´ve included the use of Date object in commit [c657b43]

File: Juliana/second-assignment/script.js (hunk 2 above)
Comment (JulianaCotabarren) [reply Juliana/second-assignment/script.js]: I´ve included the use of Date object in commit [c657b43]

File: Juliana/second-assignment/script.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (JulianaCotabarren) [reply Juliana/second-assignment/script.js]: Done in commit [ca29d60]

File: Juliana/second-assignment/script.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+loadSortedHeroesBtn.addEventListener('click', () => {
+    replaceContainer(heroesContainer, sortedHeroesContainer);
+    sortHeroes();
+    showHeroes(arrSortedHeroes, sortedHeroesContainer);
+    document.querySelector('#loadHeroes').style.display = 'flex';
+    document.querySelector('#loadSortedHeroes').style.display = 'none';
+});
+
+function showData(){
+    const url = './marvel.json';
```
Comment (mhechavarria) [inline Juliana/second-assignment/script.js:27]: Sorry @JulianaCotabarren, I've just noticed that you are using the JSON inside your folder, please use the one in the root mock folder so we don't have duplicate files.

Comment (JulianaCotabarren) [reply Juliana/second-assignment/script.js]: Done in commit [ac3270d]

**Review by mhechavarria — APPROVED** [review]
LGTM, great job!

File: Juliana/second-assignment/styles.css [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+    flex-direction: column;
+    justify-content: center;
+    margin-left: 30px;
+    padding-top: 80px;
+    width: 500px;
+    height: 92vh;
+    max-width: 80%;
+}
+#intro h1,p{
+    display: flex;
```
Comment (marianowadi) [inline Juliana/second-assignment/styles.css:78]: Normally, we leave flex properties for presentational containers (divs and the semantic ones), and just let the flex properties of those parent containers determine the alignment of their flexbox children. #intro div could have a align-items: center; property to align its content

Comment (JulianaCotabarren) [reply Juliana/second-assignment/styles.css]: I've included the flex properties as suggested in commit [0876d3b]

**Review by marianowadi — APPROVED** [review]
Great job!

**Review by mhechavarria — APPROVED** [review]
LGTM x2

**Pull request merged**